experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -399,14 +399,16 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 	}
 
 	/**
-	 * Rebase the changes that have been applied to this branch over all the divergent changes in the given branch.
+	 * Rebase the changes that have been applied to this branch over divergent changes in the given branch.
 	 * After this operation completes, this branch will be based off of `branch`.
 	 * @param branch - the branch to rebase onto
+	 * @param upTo - the furthest commit on `branch` over which to rebase (inclusive). Defaults to the head commit of `branch`.
 	 * @returns the net change to this branch and the commits that were removed and added to this branch by the rebase,
 	 * or undefined if nothing changed
 	 */
 	public rebaseOnto(
 		branch: SharedTreeBranch<TEditor, TChange>,
+		upTo = branch.getHead(),
```

Comment:
I'm not sure how I feel about this as a user facing API.  Is this something we actually want to allow or is it just necessary for peer branches?

Reply:
Very good question. I agree this shouldn't be user facing, it's a little bit too much. Thankfully, it's not - `SharedTreeBranch` is not exposed in our API. `SharedTreeView` is, and I also would not feel good about adding this level of flexibility there without a good reason. But I do think this is a generally useful degree of freedom for us internally.